src/backend/columnar/columnar_metadata.c

```diff
@@ -218,6 +219,117 @@ InitColumnarOptions(Oid regclass)
 }
 
 
+/*
+ * ExtractColumnarOptions - split inoutOptions into two lists, one for
+ * columnar options and another for everything else.
+ */
+List *
+ExtractColumnarRelOptions(List *inOptions, List **outColumnarOptions)
+{
+	List *columnarOptions = NIL;
+	List *otherOptions = NIL;
+	ListCell *lc;
+
+	foreach (lc, inOptions)
+	{
+		DefElem *elem = castNode(DefElem, lfirst(lc));
+
+		if (elem->defnamespace != NULL &&
+			strcmp(elem->defnamespace, COLUMNAR_RELOPTION_NAMESPACE) == 0)
+		{
+			columnarOptions = lappend(columnarOptions, elem);
+		}
+		else
+		{
+			otherOptions = lappend(otherOptions, elem);
+		}
+	}
+
+	*outColumnarOptions = columnarOptions;
+
+	return otherOptions;
+}
+
+
+/*
+ * SetColumnarRelOptions - apply the list of DefElem options to the relation.
+ */
+void
+SetColumnarRelOptions(RangeVar *rv, List *reloptions)
+{
+	ColumnarOptions options;
```

Comment:
we tend to zero initialize all variables

```suggestion
	ColumnarOptions options = { 0 };
```

Reply:
Done.